datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -950,12 +949,6 @@ fn add_hash_on_top(
         //   requirements.
         // - Usage of order preserving variants is not desirable (per the flag
         //   `config.optimizer.prefer_existing_sort`).
-        if repartition_beneficial_stats {
-            // Since hashing benefits from partitioning, add a round-robin repartition
-            // before it:
-            input = add_roundrobin_on_top(input, n_target)?;
-        }
-
```

Comment:
This `add_roundrobin_on_top` currently is not controlled by `enable_round_robin`.